### DISCUSSION TITLE
'"react"' has no exported member named 'RefCallback'

### DISCUSSION THREAD

**Discussion by Schmell** [chakra-ui/chakra-ui#6674] [Q&A]
I have been trying to switch my project to Vite + preact + ts (was just preactCLI + ts)
I have fixed my previous problems by aliasing react path in preact :)
Now i have these  errors to tackle 
`
node_modules/@chakra-ui/descendant/src/use-descendant.ts:2:10 - error TS2724:
 '"react"' has no exported member named 'RefCallback'. Did you mean 'useCallback'?`
I am assuming that because i am aliasing react to preact that preact/compat does not export RefCallback

I also get this
`node_modules/@chakra-ui/descendant/src/use-descendant.ts:11:56 - error TS2344: Type 'K' does not satisfy the constraint 'Record<string, any>'.
11   const descendants = useRef(new DescendantsManager<T, K>())`
in fact the K generic has errors throughout the file I don't know enough about typescript to understand it all, but can be fixed by adding `K extends Record<string, any>` as suggested by vscode's quickfix, so i suppose i can fork chakra and fix this file, but that is not ideal and I would prefer not to 

I am just about to give up on preact or chakra completely, but i don't really feel thrilled about refactoring every file in my project right now

**Comment by boralyl** [comment]
If you are using 1.x.x you will need to upgrade to v1.8.9.  See https://github.com/chakra-ui/chakra-ui/issues/6671 This will fix your 2nd problem.  

To fix the first @types/react must be version `"^16.9.23"`
